### PR TITLE
Make platform for SC parameter parsing.

### PR DESCRIPTION
**Pull request by humblec**
**Describe what this PR does**
Provide some context for the reviewer

**Is there anything that requires special attention?**
Do you have any questions? Did you do something clever?

**Related issues:**
Mention any github issues relevant to this PR

### CODE REVIEW

**Comment by humblec** [comment]
@JohnStrunk @Madhu-1 PTAL. The goal of this patch is to place hold the way we can parse SC parameters.

File: pkg/glusterfs/controllerserver.go [hunk 1]
```diff
@@ -59,11 +60,27 @@ func (cs *ControllerServer) ParseCreateVolRequest(req *csi.CreateVolumeRequest)
...
 		reqConf.gdVolReq.Name = req.Name
 	}
+
+	for k, v := range req.GetParameters() {
+
+		switch strings.ToLower(k) {
+
+		// Placeholder for more SC parameters.
+		case "snapshotenabled":
+			snapshotEnabled = strings.ToLower(v) == "true"
```
Comment (Madhu-1) [inline pkg/glusterfs/controllerserver.go]: the validation for the `v` value is missing here, if user sets some junk value for `snapshotenabled`  we need to fail the volume create request

Comment (Madhu-1) [inline pkg/glusterfs/controllerserver.go]: Validation for the v value is missing here, if user sets some junk value for `snapshotenabled` we need to fail the volume create request

Comment (Madhu-1) [inline pkg/glusterfs/controllerserver.go]: Validation for the v value is missing here, if user sets some junk value for `snapshotenabled` we need to fail the volume create request

File: pkg/glusterfs/controllerserver.go [hunk 2]
```diff
@@ -59,11 +60,27 @@ func (cs *ControllerServer) ParseCreateVolRequest(req *csi.CreateVolumeRequest)
...
 	if req != nil {
 		reqConf.gdVolReq.Name = req.Name
 	}
+
+	for k, v := range req.GetParameters() {
+
+		switch strings.ToLower(k) {
+
+		// Placeholder for more SC parameters.
+		case "snapshotenabled":
```
Comment (Madhu-1) [inline pkg/glusterfs/controllerserver.go]: Validation for the v value is missing here if the user sets some junk value for `snapshotenabled` we need to fail the volume create request

Comment (Madhu-1) [inline pkg/glusterfs/controllerserver.go]: Validation for the v value is missing here if the user sets some junk value for `snapshotenabled` we need to fail the volume create request

Comment (Madhu-1) [inline pkg/glusterfs/controllerserver.go]: Validation for the v value is missing here if the user sets some junk value for `snapshotenabled` we need to fail the volume create request

Comment (Madhu-1) [inline pkg/glusterfs/controllerserver.go]: Validation for the v value is missing here if the user sets some junk value for `snapshotenabled` we need to fail the volume create request

Comment (Madhu-1) [inline pkg/glusterfs/controllerserver.go]: Validation for the v value is missing here if the user sets some junk value for `snapshotenabled` we need to fail the volume create request

Comment (Madhu-1) [inline pkg/glusterfs/controllerserver.go]: Validation for the v value is missing here if the user sets some junk value for `snapshotenabled` we need to fail the volume create request

Comment (Madhu-1) [inline pkg/glusterfs/controllerserver.go]: Validation for the v value is missing here if the user sets some junk value for `snapshotenabled` we need to fail the volume create request

Comment (Madhu-1) [inline pkg/glusterfs/controllerserver.go]: Validation for the v value is missing here, if user sets some junk value for `snapshotenabled` we need to fail the volume create request

Comment (Madhu-1) [inline pkg/glusterfs/controllerserver.go]: Validation for the v value is missing here, if user sets some junk value for `snapshotenabled` we need to fail the volume create request

Comment (Madhu-1) [inline pkg/glusterfs/controllerserver.go]: Validation for the v value is missing here, if user sets some junk value for `snapshotenabled` we need to fail the volume create request

Comment (Madhu-1) [inline pkg/glusterfs/controllerserver.go]: Validation for the v value is missing here, if user sets some junk value for `snapshotenabled` we need to fail the volume create request

Comment (humblec) [reply pkg/glusterfs/controllerserver.go]: `!true` is considered `false`. That should be fine, isnt it @Madhu-1

File: pkg/glusterfs/controllerserver.go [hunk 3]
```diff
@@ -59,11 +60,27 @@ func (cs *ControllerServer) ParseCreateVolRequest(req *csi.CreateVolumeRequest)
...
+			snapshotEnabled = strings.ToLower(v) == "true"
+
+		default:
+			return nil, fmt.Errorf("invalid option %q given for glusterfs CSI driver %s", k, glusterfsCSIDriverName)
```
[the remark this answers is not in the document]
Comment (humblec) [reply pkg/glusterfs/controllerserver.go]: @Madhu-1 Done.

**Comment by humblec** [comment]
@JohnStrunk PTAL .

**Comment by humblec** [comment]
@JohnStrunk any comments on this PR ? :)

**Comment by JohnStrunk** [comment]
Let's hold this until we get actual parameters to parse.

Also, validation is important. `snapshotEnabled: yes` would neither produce an error, nor the expected behavior... same is the case with a simple misspelling `snapshotEnabled: treu`.

**Pull request closed without merging**